ignite/cmd/bubblemodel/testnet_multi_node.go

```diff
@@ -0,0 +1,200 @@
+package cmdmodel
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"syscall"
+
+	tea "github.com/charmbracelet/bubbletea"
+
+	"github.com/ignite/cli/v29/ignite/services/chain"
+)
+
+type NodeStatus int
+
+const (
+	Stopped NodeStatus = iota
+	Running
+)
+
+type MultiNode struct {
+	ctx  context.Context
+	appd string
+	args chain.MultiNodeArgs
+
+	nodeStatuses []NodeStatus
+	pids         []int      // Store the PIDs of the running processes
+	numNodes     int        // Number of nodes
+	logs         [][]string // Store logs for each node
+}
+
+type ToggleNodeMsg struct {
+	nodeIdx int
+}
+
+type UpdateStatusMsg struct {
+	nodeIdx int
+	status  NodeStatus
+}
+
+type UpdateLogsMsg struct{}
+
+func UpdateDeemon() tea.Cmd {
+	return func() tea.Msg {
+		return UpdateLogsMsg{}
+	}
+}
+
+// Initialize the model.
+func NewModel(ctx context.Context, chainname string, args chain.MultiNodeArgs) MultiNode {
+	numNodes, err := strconv.Atoi(args.NumValidator)
+	if err != nil {
+		panic(err)
+	}
+	return MultiNode{
+		ctx:          ctx,
+		appd:         chainname + "d",
+		args:         args,
+		nodeStatuses: make([]NodeStatus, numNodes), // initial states of nodes
+		pids:         make([]int, numNodes),
+		numNodes:     numNodes,
+		logs:         make([][]string, numNodes), // Initialize logs for each node
+	}
+}
+
+// Implement the Update function.
+func (m MultiNode) Init() tea.Cmd {
+	return nil
+}
+
+// ToggleNode toggles the state of a node.
+func ToggleNode(nodeIdx int) tea.Cmd {
+	return func() tea.Msg {
+		return ToggleNodeMsg{nodeIdx: nodeIdx}
+	}
+}
+
+// Run or stop the node based on its status.
+func RunNode(nodeIdx int, start bool, m MultiNode) tea.Cmd {
+	var (
+		pid  = &m.pids[nodeIdx]
+		args = m.args
+		appd = m.appd
+	)
+
+	return func() tea.Msg {
+		if start {
+			nodeHome := filepath.Join(args.OutputDir, args.NodeDirPrefix+strconv.Itoa(nodeIdx))
+			// Create the command to run in background as a daemon
+			cmd := exec.Command(appd, "start", "--home", nodeHome)
```

Comment:
IMHO, we should exec inside a goroutine; we can use `errgroup` or something similar. But using PID can easily break the logic